### PR TITLE
docs: document bootstrap stages

### PR DESCRIPTION
**Pull request by pothos**
The bootstrap stages, example outputs and their meaning is documented.

### CODE REVIEW

File: docs/usage/bootstrap.md [hunk 1]
```diff
@@ -276,3 +276,36 @@ If some nodes should be excluded upfront from provisioning, maybe because they a
...
 
+## Bootstrap Stages
+
+Racker displays the progress of the different stages by updating the terminal output.
+The first stage checks whether all BMCs are rechable through IPMI.
+When encountering failures, the process should be restarted, possibly with excluding problematic nodes through the `-exclude` flag.
+The second stage checks that the initial OS installation via PXE completes.
+On failure, it offers to retry or exclude problematic nodes depending on the `-onfailure` flag.
+
+Here an example for a Flatcar Container Linux provisioning that automatically excludes two nodes:
```
Comment (joaquimrocha) [inline docs/usage/bootstrap.md:287]:
```suggestion
Here is an example of a Flatcar Container Linux provisioning that automatically excluded two nodes:
```

File: docs/usage/bootstrap.md [hunk 2]
```diff
@@ -276,3 +276,36 @@ If some nodes should be excluded upfront from provisioning, maybe because they a
...
+➤ OS installation via PXE (8/10)... × failed
+Failed to provision the following 2 nodes:
+aa:bb:cc:dd:ee:11 aa:bb:cc:dd:ee:22
+You can see logs in /home/core/flatcar-container-linux/logs/2021-05-07_17-11-59, run 'ipmi <MAC|DOMAIN> diag' for a short overview of a node, connect to the serial console via 'ipmi <MAC|DOMAIN>', or try to connect via SSH.
+Something went wrong, removing 2 nodes from config and retrying 1/3
+➤ OS installation via PXE (8/8)... ✓ done
+```
+
+For Lokomotive there are additional stages. First the Kubernetes API bring-up, then the cluster health check, and finally the Lokomotive Component installation.
+When some nodes to not join the cluster, the health check stage offers to retry or exclude problematic nodes depending on the `-onfailure` flag.
```
Comment (joaquimrocha) [inline docs/usage/bootstrap.md:300]:
```suggestion
When some nodes do not join the cluster, the health check stage offers to retry or exclude problematic nodes depending on the `-onfailure` flag.
```

File: docs/usage/bootstrap.md [hunk 3]
```diff
@@ -276,3 +276,36 @@ If some nodes should be excluded upfront from provisioning, maybe because they a
...
+Even with all nodes present the cluster health check may still fail due to other reasons.
+
+Here an example for a completely successful Lokomotive provisioning:
```
Comment (joaquimrocha) [inline docs/usage/bootstrap.md:303]:
```suggestion
Here is an example of a completely successful Lokomotive provisioning:
```

**Review by joaquimrocha — APPROVED** [review]
A few corrections, otherwise LGTM.

**Comment by pothos** [comment]
Thanks!

**Pull request merged**